lib/wallet/rpc.js

```diff
@@ -1912,61 +1998,149 @@ class RPC extends RPCBase {
     if (bid > value)
       throw new RPCError(errs.TYPE_ERROR, 'Invalid bid.');
 
-    const tx = await wallet.sendBid(name, bid, value);
-
-    return tx.getJSON(this.network);
+    return {
+      name,
+      bid,
+      value,
+      account
+    };
   }
 
   async sendReveal(args, help) {
-    if (help || args.length > 1)
-      throw new RPCError(errs.MISC_ERROR, 'sendreveal "name"');
-
+    const opts = this._validateReveal(args, help, 'sendreveal');
     const wallet = this.wallet;
-    const valid = new Validator(args);
-    const name = valid.str(0);
 
-    if (!name) {
+    if (!opts.name) {
```

Comment:
What do you think about having a specific RPC for `sendrevealall` and `createrevealall` so that a bug doesn't cause the wallet to reveal everything when it was not intended to?

Reply:
I think this is a great idea, but out of scope for this PR.